batch2/batch/batch.py

```diff
@@ -227,6 +229,58 @@ def getopt(obj, attr):
 
     assert gsa_key
 
+    service_account_name = job_spec.get('service_account_name')
+    if service_account_name:
+        sa = await k8s_client.read_namespaced_service_account(
+            service_account_name, BATCH_NAMESPACE,
+            _request_timeout=KUBERNETES_TIMEOUT_IN_SECONDS)
+        assert len(sa.secrets) == 1
+
+        token_secret_name = sa.secrets[0].name
+
+        secret = await k8s_client.read_namespaced_secret(
+            token_secret_name, BATCH_NAMESPACE,
+            _request_timeout=KUBERNETES_TIMEOUT_IN_SECONDS)
+
+        token = base64.b64decode(secret.data['token']).decode()
+        namespace = base64.b64decode(secret.data['namespace']).decode()
+        cert = secret.data['ca.crt']
+
+        kube_config = f'''
+apiVersion: v1
+clusters:
+- cluster:
+    certificate-authority: /.kube/ca.crt
+    server: {KUBERNETES_SERVER_URL}
+  name: default-cluster
+contexts:
+- context:
+    cluster: default-cluster
+    user: {service_account_name}
+    namespace: {namespace}
+  name: default-context
+current-context: default-context
+kind: Config
+preferences: {{}}
+users:
+- name: {service_account_name}
+  user:
+    token: {token}
+'''
+
+        job_spec['secrets'].append({
+            'namespace': 'batch-pods',  # FIXME unused
+            'name': 'kube-config',
+            'mount_path': '/.kube',
+            'mount_in_copy': False,
```

Comment:
This is the default.

 >                 if secret.get('mount_in_copy', False):